### PR TITLE
Add dockerfile for CI runners

### LINKED ISSUE #185: [FEA] Create a new base image for rapids ci
**Is your feature request related to a problem? Please describe.**
Currently we are using `gpuci/rapidsai-driver:21.10-cuda11.4-devel-ubuntu20.04-py3.8` which is 14GB in size, however the newer `rapidsai/ci:cuda11.4.1-ubuntu20.04-py3.8` image is only 1.5GB if we add a few more needed packages into that base we could limit the amount of time we spend in I/O for CI.

**Describe the solution you'd like**
New image.

### PR DESCRIPTION
**Pull request by dagardner-nv**
Builds two images a `srf-ci` image and a `srf-ci-driver` variant.
Image sizes are 6GB and 6.1GB respectively (current image being used in CI is  14GB) 

CI usage drops from around an hour and a half (https://github.com/nv-morpheus/SRF/actions/runs/3357911701/usage) down to about 40 minutes (https://github.com/nv-morpheus/SRF/actions/runs/3372276756)

Image contains:
+ Pre-populated conda env
+ IWYU
+ Nvidia libs needed to build

fixes #185

### CODE REVIEW

File: ci/runner/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        llvm-${CLANG_VER}-dev \
+    && rm -rf /var/lib/apt/lists/* &&\
+    update-alternatives --install /usr/bin/clang clang /usr/bin/clang-${CLANG_VER} 121 \
+        --slave /usr/bin/clang-apply-replacements clang-apply-replacements /usr/bin/clang-apply-replacements-${CLANG_VER} \
+        --slave /usr/bin/clang-format clang-format /usr/bin/clang-format-${CLANG_VER} \
+        --slave /usr/bin/clang-format-diff clang-format-diff /usr/bin/clang-format-diff-${CLANG_VER} \
+        --slave /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-${CLANG_VER} \
+        --slave /usr/bin/clang-tidy-diff.py clang-tidy-diff.py /usr/bin/clang-tidy-diff-${CLANG_VER}.py \
+        --slave /usr/bin/clang++ clang++ /usr/bin/clang++-${CLANG_VER} \
+        --slave /usr/bin/git-clang-format git-clang-format /usr/bin/git-clang-format-${CLANG_VER}
```
Comment (mdemoret-nv) [inline ci/runner/Dockerfile:55]: We should move to using clang installed via conda. Would close #163 if clang and IWYU were all installed via conda

File: ci/runner/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    cd /tmp/iwyu &&\
+    git checkout clang_${CLANG_VER} &&\
+    source activate ${PROJ_NAME} && \
+    cmake -G Ninja \
+        -DCMAKE_C_COMPILER=$(which clang) \
+        -DCMAKE_CXX_COMPILER=$(which clang++) \
+        -DCMAKE_INSTALL_PREFIX=${CONDA_PREFIX}  . && \
+    cmake --build . --parallel $(nproc) --target install && \
+    cd - && \
+    rm -rf /tmp/iwyu
```
Comment (mdemoret-nv) [inline ci/runner/Dockerfile:77]: Same comment here. We can install this now via conda.

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
Couple of suggestions.

File: ci/runner/Dockerfile (hunk 1 above)
Comment (dagardner-nv) [reply ci/runner/Dockerfile:55]: I tried both the apt version of clang and the conda package. The apt version is I believe 500MB smaller, so I went with it.

File: ci/runner/Dockerfile (hunk 2 above)
Comment (dagardner-nv) [reply ci/runner/Dockerfile:77]: The version in conda requires clang-14, which we aren't on until Ryan's PR #156 is merged.

**Comment by mdemoret-nv** [comment]
@dagardner-nv Do you know why the code coverage dropped so significantly with this PR?

**Review by mdemoret-nv — APPROVED** [review]
Approving these changes now and we will look into the coverage issues in issue #211

**Comment by mdemoret-nv** [comment]
@gpucibot merge

**Pull request merged**